fedimint-api-client/src/api.rs

```diff
@@ -1117,29 +1118,35 @@ pub struct WsFederationApi<C = WsClient> {
     module_id: Option<ModuleInstanceId>,
 }
 
-/// Some data shared/preserved between [`FederationPeerClient`] and
+/// Connection state shared/preserved between [`FederationPeerClient`] and the
 /// Jit tasks it spawns.
 #[derive(Debug)]
-struct FederationPeerClientShared {
-    last_connection_attempt: SystemTime,
-    connection_attempts: u64,
+struct FederationPeerClientConnectionState {
+    /// Last time a connection attempt was made, or `None` if no attempt has
+    /// been made yet.
+    last_connection_attempt_or: Option<SystemTime>,
+    connection_backoff: backon::FibonacciBackoff,
 }
 
-impl FederationPeerClientShared {
+impl FederationPeerClientConnectionState {
+    const MIN_BACKOFF: Duration = Duration::from_millis(100);
+    const MAX_BACKOFF: Duration = Duration::from_secs(5);
```

Comment:
In #5460 :

> t seems it is linear with a max wait time of 5 seconds. 

Was the desire only to make it exponential or to increase the max as well? @benthecarman thoughts?